repa/grids/diffusion.cpp

```diff
@@ -391,5 +365,72 @@ void Diffusion::init_new_foreign_cell(local_cell_index_type localcell,
     util::push_back_unique(borderCellsNeighbors[localcell], owner);
 }
 
+void Diffusion::command(std::string s)
+{
+    std::smatch m;
+
+    static const std::regex profit_re(
+        "(set) (profit) (pass) (through) (([[:digit:]]*[.])?[[:digit:]]+)");
+    if (std::regex_match(s, m, profit_re)) {
+        double ppt = std::stod(m[5].str().c_str(), NULL);
+        if (comm_cart.rank() == 0)
+            std::cout << "Setting profit pass through = " << ppt << std::endl;
+        profit_percentage_pass_through = ppt;
+        return;
+    }
+
+    static const std::regex iter_re("(set) (flow_count) ([[:digit:]]+)");
+    if (std::regex_match(s, m, iter_re)) {
+        uint32_t flow_count = std::stoul(m[3].str().c_str(), NULL);
+        if (comm_cart.rank() == 0)
+            std::cout << "Setting flow_count = " << flow_count << std::endl;
+        try {
+            DIFFUSION_MAYBE_SET_NFLOW_ITER(flow_calc.get(), flow_count);
+        }
+        catch (...) {
+            std::cerr << "Cannot set nflow iter." << std::endl;
+        }
+        return;
+    }
+
+    static const std::regex beta_re(
+        "(set) (beta) (([[:digit:]]*[.])?[[:digit:]]+)");
+    if (std::regex_match(s, m, beta_re)) {
+        double beta_value = std::stod(m[3].str().c_str(), NULL);
+        if (comm_cart.rank() == 0)
+            std::cout << "Setting beta = " << beta_value << std::endl;
+        try {
+            DIFFUSION_MAYBE_SET_BETA(flow_calc.get(), beta_value);
+        }
+        catch (...) {
+            std::cerr << "Cannot set nflow iter." << std::endl;
+        }
+        return;
+    }
+
+    static const std::regex flow_re(
+        "(set) (flow) (willebeek|schornbaum|so|sof)");
+    if (std::regex_match(s, m, flow_re)) {
+        const std::string &impl = m[3].str();
+        if (comm_cart.rank() == 0)
+            std::cout << "Setting implementation to: " << impl << std::endl;
+        if (impl == "willebeek")
+            flow_calc = diff_variants::create_flow_calc(
+                diff_variants::FlowCalcKind::WILLEBEEK);
+        else if (impl == "schornbaum")
+            flow_calc = diff_variants::create_flow_calc(
+                diff_variants::FlowCalcKind::SCHORN);
+        else if (impl == "so")
+            flow_calc = diff_variants::create_flow_calc(
+                diff_variants::FlowCalcKind::SO);
+        else if (impl == "sof")
+            flow_calc = diff_variants::create_flow_calc(
+                diff_variants::FlowCalcKind::SOF);
+        else
+            assert(false);
```

Comment:
Better to print an error here that "`impl' is unknown".

Reply:
Also fixed in the mentioned patch: https://github.com/hirschsn/repa/pull/36#discussion_r408405605

Reply:
Updated in 337159c.

Currently an error is output and the program is terminated in the debug variant. 
In the release variant, the default variant (Willebeek) would be used here. We could consider whether we really want to do this, or whether we should end the program here as well, since the user has chosen a different implementation and may have made a typo.

Reply:
Yes, using the default implementation and issuing a warning is good.